clientgenv2/source_generator.go

```diff
@@ -76,6 +69,97 @@ func (rs ResponseFieldList) IsStructType() bool {
 	return len(rs) > 0 && !rs.IsFragment()
 }
 
+type StructGenerator struct {
+	currentResponseFieldList *ResponseFieldList
+	preMergedStructSources   []*StructSource
+	postMergedStructSources  []*StructSource
+}
+
+func NewStructGenerator(responseFieldList ResponseFieldList) *StructGenerator {
+	currentFields := make(ResponseFieldList, 0)
+	fragmentChildrenFields := make(ResponseFieldList, 0)
+	for _, field := range responseFieldList {
+		if field.IsFragmentSpread {
+			fragmentChildrenFields = append(fragmentChildrenFields, field.ResponseFields...)
+		} else {
+			currentFields = append(currentFields, field)
+		}
+	}
+
+	preMergedStructSources := make([]*StructSource, 0)
+	postMergedStructSources := make([]*StructSource, 0)
+
+	for _, field := range responseFieldList {
+		if field.IsFragmentSpread {
+			preMergedStructSources = append(preMergedStructSources, &StructSource{
+				Name: field.FieldTypeString(),
+				Type: field.ResponseFields.StructType(),
+			})
+		}
+	}
+
+	currentFields = mergeFieldsRecursively(currentFields, fragmentChildrenFields, &preMergedStructSources, &postMergedStructSources)
+	return &StructGenerator{
+		currentResponseFieldList: &currentFields,
+		preMergedStructSources:   preMergedStructSources,
+		postMergedStructSources:  postMergedStructSources,
+	}
+}
+
+func mergeFieldsRecursively(targetFields ResponseFieldList, sourceFields ResponseFieldList, preMerged, postMerged *[]*StructSource) ResponseFieldList {
+	for _, sourceField := range sourceFields {
+		sameNameFieldFlag := false
+		for _, targetField := range targetFields {
+			if sourceField.Name == targetField.Name && !targetField.ResponseFields.IsBasicType() {
```

Comment:
With the current implementation, it seems like there are unnecessary loops, and the nesting feels a bit excessive. I’d like to simplify this by using the following implementation with a map:
```
func (rs ResponseFieldList) MapByName() map[string]*ResponseField {
	m := make(map[string]*ResponseField)
	for _, v := range rs {
		m[v.Name] = v
	}
	return m
}
```
This approach should make the logic cleaner and more efficient.

Reply:
It's true, I fixed this.